packages/beacon-node/src/network/peers/discover.ts

```diff
@@ -267,10 +268,17 @@ export class PeerDiscovery {
   }
 
   /**
-   * Progressively called by libp2p peer discovery as a result of any query.
+   * Progressively called by libp2p as a result of peer discovery or updates to its peer store
    */
   private onDiscoveredPeer = (evt: CustomEvent<PeerInfo>): void => {
     const {id, multiaddrs} = evt.detail;
+
+    // libp2p may send us PeerInfos without multiaddrs
+    if (multiaddrs.length === 0) {
```

Comment:
with this error `Cannot read properties of undefined (reading '0')` we need to handle undefined `multiaddrs` too

```suggestion
    if (!multiaddrs || multiaddrs.length === 0) {
```